File: 10828.py
from sys import stdin
def push(stack, n):
    stack.append(n)

def pop(stack):
    if len(stack) == 0:
        return -1
    return stack.pop()

def size(stack):
    return len(stack)

def empty(stack):
    return int(len(stack) == 0)

def top(stack):
    if empty(stack):
        return -1
    return stack[len(stack)-1]

stck = []
cmdNum = int(input())
for _ in range(cmdNum):
    tmp = stdin.readline().split()

    if tmp[0] == "push":
        push(stck, int(tmp[1]))
    elif tmp[0] == "pop":
        print(pop(stck))
    elif tmp[0] == "size":
        print(size(stck))
    elif tmp[0] == "empty":
        print(empty(stck))
    elif tmp[0] == "top":
        print(top(stck))

File: baekJoon/10828.cpp
#include <iostream>
#include <stdlib.h>
#include <string>

using namespace std;

class stack{
    int pos;
    int data[10000];
public:
    stack(){
        pos = 0;
    }
    void push(int n){
        data[pos++] = n;
    }
    int pop(){
        if(pos == 0) return -1;
        return data[--this->pos];
    }
    int size(){
        return pos;
    }
    int empty(){
        return (pos==0);
    }
    int top(){
        if(pos ==0) return -1;
        return data[pos-1];
    }
};

int main(){
    string cmd;
    int n;
    stack s;
    int cmdNum;
    cin >> cmdNum;
    for(int i=0; i<cmdNum; i++){
        cin >> cmd;
        if (!cmd.compare("push")) {
            cin >> n;
            s.push(n);
        }   
        else if(!cmd.compare("pop")){cout << s.pop() << endl;}
        else if(!cmd.compare("empty")){cout<<1*s.empty()<<endl;}
        else if(!cmd.compare("top")){cout << s.top() << endl;}
        else if(!cmd.compare("size")){cout << s.size() << endl;}
    }
    return 0;
}